aiida/tools/importexport/dbexport/utils.py

```diff
@@ -403,171 +402,39 @@ def retrieve_linked_nodes(process_nodes, data_nodes, **kwargs):  # pylint: disab
-                    link_type_value=LinkType.CALL_WORK.value
-                )
-                process_nodes.update(found_nodes - retrieved_nodes)
-                links_uuid_dict.update(links_uuids)
-
-        # If it is a Data
-        else:
-            current_node_pk = data_nodes.pop()
-            # If it is already visited continue to the next node
-            if current_node_pk in retrieved_nodes:
-                continue
-            # Otherwise say that it is a node to be exported
-            else:
-                retrieved_nodes.add(current_node_pk)
-
-            # INPUT_CALC(Data, CalculationNode) - Forward
-            if traversal_rules['input_calc_forward']:
-                links_uuids, found_nodes = _retrieve_linked_nodes_query(
-                    current_node_pk,
-                    input_type=Data,
-                    output_type=ProcessNode,
-                    direction='forward',
-                    link_type_value=LinkType.INPUT_CALC.value
-                )
-                process_nodes.update(found_nodes - retrieved_nodes)
-                links_uuid_dict.update(links_uuids)
-
-            # CREATE(CalculationNode, Data) - Backward
-            if traversal_rules['create_backward']:
-                links_uuids, found_nodes = _retrieve_linked_nodes_query(
-                    current_node_pk,
-                    input_type=ProcessNode,
-                    output_type=Data,
-                    direction='backward',
-                    link_type_value=LinkType.CREATE.value
-                )
-                process_nodes.update(found_nodes - retrieved_nodes)
-                links_uuid_dict.update(links_uuids)
-
-            # RETURN(WorkflowNode, Data) - Backward
-            if traversal_rules['return_backward']:
-                links_uuids, found_nodes = _retrieve_linked_nodes_query(
-                    current_node_pk,
-                    input_type=ProcessNode,
-                    output_type=Data,
-                    direction='backward',
-                    link_type_value=LinkType.RETURN.value
-                )
-                process_nodes.update(found_nodes - retrieved_nodes)
-                links_uuid_dict.update(links_uuids)
-
-            # INPUT_WORK(Data, WorkflowNode) - Forward
-            if traversal_rules['input_work_forward']:
-                links_uuids, found_nodes = _retrieve_linked_nodes_query(
-                    current_node_pk,
-                    input_type=Data,
-                    output_type=ProcessNode,
-                    direction='forward',
-                    link_type_value=LinkType.INPUT_WORK.value
-                )
-                process_nodes.update(found_nodes - retrieved_nodes)
-                links_uuid_dict.update(links_uuids)
-
-    return retrieved_nodes, list(links_uuid_dict.values()), traversal_rules
+    # Creating a set of pks that are the process nodes and data nodes:
+    starting_nodes = process_nodes.union(data_nodes)
```

Comment:
If the distinction between process and data nodes is no longer required, we may want to rewrite the `export_tree` function accordingly.
But this is for a follow-up PR.